prover/circuits/pi-interconnection/assign.go

```diff
@@ -287,7 +287,7 @@ func (c *Compiled) Assign(r Request) (a Circuit, err error) {
 			// @alex: Not sure this check is a duplicate because we already check
 			// that the state root hash is well-propagated and this should be
 			// enough that the rolling hash update events are emitted in sequence.
-			if executionFPI.FirstRollingHashUpdateNumber != lastRollingHashNumber+1 {
+			if executionFPI.FirstRollingHashUpdateNumber >= lastRollingHashNumber {
```

Comment:
@Tabaie as a FYI

Reply:
@thedarkjester confirmed that the rolling hash was updated in batch. So the consecutivity check is moot here.

Reply:
1. Shouldn't it be `executionFPI.FirstRollingHashUpdateNumber <= lastRollingHashNumber` instead then? The if checks for what _shouldn't_ happen iirc.
2. Is the change also reflected on the circuit side?

Reply:
Oh right, it has not been committed. Just addressed!